externals/kyuubi-flink-sql-engine/src/main/scala/org/apache/kyuubi/engine/flink/FlinkSQLEngine.scala

```diff
@@ -72,6 +75,20 @@ object FlinkSQLEngine extends Logging {
     try {
       val flinkConfDir = CliFrontend.getConfigurationDirectoryFromEnv
       val flinkConf = GlobalConfiguration.loadConfiguration(flinkConfDir)
+
+      val executionTarget = flinkConf.getString(DeploymentOptions.TARGET)
+      // set cluster name for per-job and application mode
+      executionTarget match {
+        case "yarn-per-job" | "yarn-application" =>
+          val appName = s"kyuubi_${user}_flink_${Instant.now}"
+          flinkConf.setString("yarn.application.name", appName)
+        case "kubernetes-application" =>
+          val appName = s"kyuubi-${user}-flink-${Instant.now}"
+          flinkConf.setString("kubernetes.cluster-id", appName)
+        case null =>
+          error("No execution.target specified for Flink engine")
+      }
```

Comment:
we need a `case _ =>` branch to handle other cases. BTW, can `executionTarget` be `null`?

Reply:
Oh, I thought we could bypass session clusters and standalone clusters because they already have cluster names, so what a `case _ =>` branch can do is simply validate the values. Is this what you expect? 

> BTW, can executionTarget be `null`?

The default value is `null`, which would cause a validation error on job submission.

Reply:
if no `case _ =>` branch, the program will crash when receive the unmatched value
![image](https://user-images.githubusercontent.com/26535726/150770417-846cbb16-f480-4cda-a788-202467cd1a97.png)